src/components/gridlines.ts

```diff
@@ -79,31 +171,34 @@ export class Gridlines extends Component {
 
   private _redrawXLines() {
     if (this._xScale) {
-      const xTicks = this._xScale.ticks();
-      const getScaledXValue = (tickVal: number) => this._xScale.scale(tickVal);
+      const between = this.betweenX();
+      const xTicks = this._xScale.ticks().slice(between ? 1 : 0);
       const xLinesUpdate = this._xLinesContainer.selectAll("line").data(xTicks);
       const xLines = xLinesUpdate.enter().append("line").merge(xLinesUpdate);
-      xLines.attr("x1", getScaledXValue)
+      xLines.attr("x1", gridPositionFactory(this._xScale, between, this._xScale.ticks()))
```

Comment:
I don't know how expensive `.ticks()` is but you already called it above, so you may as well re-use that